quinn-proto/src/endpoint.rs

```diff
@@ -559,15 +548,18 @@ impl Endpoint {
         match conn.handle_first_packet(now, addresses.remote, ecn, packet_number, packet, rest) {
             Ok(()) => {
                 trace!(id = ch.0, icid = %dst_cid, "connection incoming");
-                Some((ch, conn))
+                Some(DatagramEvent::NewConnection(ch, conn))
             }
             Err(e) => {
                 debug!("handshake failed: {}", e);
                 self.handle_event(ch, EndpointEvent(EndpointEventInner::Drained));
                 if let ConnectionError::TransportError(e) = e {
-                    self.initial_close(version, addresses, crypto, &src_cid, &loc_cid, e);
+                    Some(DatagramEvent::Response(self.initial_close(
```

Comment:
Style nit: maybe replace the `if let` with a `match` here?

Reply:
Done.